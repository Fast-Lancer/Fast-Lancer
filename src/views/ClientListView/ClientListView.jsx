import { useEffect, useState } from 'react'
import { Link } from 'react-router-dom'
import ClientItem from '../../components/ClientItem/ClientItem.jsx'
import Title from '../../components/Title/Title.jsx'
import { getClients } from '../../services/clients.js'

export default function ClientListView() {
  const [loading, setLoading] = useState(true)
  const [clients, setClients] = useState([])

  useEffect(() => {
    getClients().then(setClients).finally(() => setLoading(false))
  }, [])

  if(loading) return <h1>Loading...</h1>

  return (
    <div>
      <Title pageTitle='clients' pageHeader='Clients' />
      {clients.map((client) => (
        <Link key={client.id} to={`/clients/${client.id}`}>
          <ClientItem key={client.id}  client={client}  />
        </Link>
      ))}
    </div>
  )
}
