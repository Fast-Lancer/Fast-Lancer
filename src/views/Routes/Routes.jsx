import { Route, Switch } from 'react-router-dom'
import Auth from '../Auth/Auth.jsx'
import Home from '../Home/Home.jsx'
import PrivateRoute from '../../components/PrivateRoute/PrivateRoute.jsx'

export default function Routes() {
  return (
    <Switch>
      <PrivateRoute path="/projects">
        <Auth />
      </PrivateRoute>
      <Route path="/login">
        <Auth />
      </Route>
      <Route path="/signup">
        <Auth />
      </Route>
      <Route path="/">
        <Home />
      </Route>
    </Switch>
  )
}
