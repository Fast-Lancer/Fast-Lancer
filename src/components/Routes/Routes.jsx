import { Route, Switch } from 'react-router-dom'
import Auth from '../../views/Auth/Auth.jsx'
import PrivateRoute from '../PrivateRoute/PrivateRoute.jsx'
import ClientListView from '../../views/Clients/ClientListView/ClientListView.jsx'
import ClientDetailView from '../../views/Clients/ClientDetailView/ClientDetailView.jsx'
import About from '../../views/About/About.jsx'
import NewEditClient from '../../views/Clients/NewEditClient/NewEditClient.jsx'
import ProjectDetailView from '../../views/Projects/ProjectDetailView/ProjectDetailView.jsx'
import ProjectListView from '../../views/Projects/ProjectListView/ProjectListView.jsx'
import NewEditProject from '../../views/Projects/NewEditProject/NewEditProject'

export default function Routes() {
  return (
    <Switch>
      <PrivateRoute exact path='/projects/edit/:id'>
        <NewEditProject />
      </PrivateRoute>
      <PrivateRoute exact path='/projects/new'>
        <NewEditProject />
      </PrivateRoute>
      <PrivateRoute exact path='/projects/:id'>
        <ProjectDetailView />
      </PrivateRoute>
      <PrivateRoute exact path='/projects'>
        <ProjectListView />
      </PrivateRoute>
      <PrivateRoute exact path='/clients/edit/:id' >
        <NewEditClient />
      </PrivateRoute>
      <PrivateRoute exact path='/clients/newclient' >
        <NewEditClient isNew/>
      </PrivateRoute>
      <PrivateRoute exact path='/clients/:id'>
        <ClientDetailView />
      </PrivateRoute>
      <PrivateRoute exact path='/clients'>
        <ClientListView />
      </PrivateRoute>
      <Route exact path='/login'>
        <Auth />
      </Route>
      <Route exact path='/signup'>
        <Auth />
      </Route>
      <Route exact path='/about'>
        <About />
      </Route>
      <Route exact path='/'>
        <Auth />
      </Route>
    </Switch>
  )
}
