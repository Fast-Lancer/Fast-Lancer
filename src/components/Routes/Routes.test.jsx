import { screen, render } from '@testing-library/react'
import { MemoryRouter, Route, Switch } from 'react-router-dom'
import Routes from './Routes'
import ClientListView from '../../views/ClientListView/ClientListView'

it('properly renders the Routes component', () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/']}>
      <Routes />
    </MemoryRouter>
  )
  expect(container).toMatchSnapshot()
})
