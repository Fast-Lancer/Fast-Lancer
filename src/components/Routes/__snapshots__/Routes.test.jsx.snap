// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`properly renders the Routes component 1`] = `
<div>
  <div
    class="test"
  >
    <div
      class="authContent"
    >
      <div
        class="titleBody"
      >
        <ul
          class="tabnav"
        >
          <li
            class="active"
          >
            <a
              href=" "
            >
              Log In
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              ⚛︎
            </a>
          </li>
        </ul>
      </div>
      <main>
        <fieldset>
          <label>
            Email:
            <input
              value=""
            />
          </label>
          <label>
            Password:
            <input
              type="password"
              value=""
            />
          </label>
          <button>
            Log In
          </button>
          <button>
            Log In With Demo Account
          </button>
          <div
            class="linkButton"
          >
            Sign Up Instead
          </div>
        </fieldset>
      </main>
    </div>
  </div>
</div>
`;
